hardware_integration/src/can_hardware_interface.cpp

```diff
@@ -14,354 +14,327 @@
 
 #include <algorithm>
 
-std::thread *CANHardwareInterface::can_thread = nullptr;
-std::thread *CANHardwareInterface::updateCANLibPeriodicThread = nullptr;
+std::unique_ptr<std::thread> CANHardwareInterface::canThread;
+std::unique_ptr<std::thread> CANHardwareInterface::periodicUpdateThread;
 std::condition_variable CANHardwareInterface::threadConditionVariable;
-std::vector<CANHardwareInterface::CanHardware *> CANHardwareInterface::hardwareChannels;
+std::vector<std::unique_ptr<CANHardwareInterface::CANHardware>> CANHardwareInterface::hardwareChannels;
 std::vector<CANHardwareInterface::RawCanMessageCallbackInfo> CANHardwareInterface::rxCallbacks;
-std::vector<CANHardwareInterface::CanLibUpdateCallbackInfo> CANHardwareInterface::canLibUpdateCallbacks;
+std::vector<CANHardwareInterface::CanLibUpdateCallbackInfo> CANHardwareInterface::periodicUpdateCallbacks;
 std::mutex CANHardwareInterface::hardwareChannelsMutex;
 std::mutex CANHardwareInterface::threadMutex;
-std::mutex CANHardwareInterface::rxCallbackMutex;
-std::mutex CANHardwareInterface::canLibNeedsUpdateMutex;
-std::mutex CANHardwareInterface::canLibUpdateCallbacksMutex;
-bool CANHardwareInterface::threadsStarted = false;
-bool CANHardwareInterface::canLibNeedsUpdate = false;
-std::uint32_t CANHardwareInterface::canLibUpdatePeriod = CANLIB_UPDATE_RATE;
-CANHardwareInterface CANHardwareInterface::CAN_HARDWARE_INTERFACE;
+std::mutex CANHardwareInterface::rxCallbacksMutex;
+std::mutex CANHardwareInterface::periodicUpdateCallbacksMutex;
+std::atomic_bool CANHardwareInterface::threadsStarted = false;
+std::atomic_bool CANHardwareInterface::canLibNeedsUpdate = false;
```

Comment:
`= false` isn't portable and won't work on with GCC because it would cause a copy to be created. `= { false }` is portable.